src/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# Source files
+file(GLOB_RECURSE SOURCES_SC2LADDERSERVER "*.cpp" "*.h")
+
+# Include directories
+include_directories(SYSTEM
+    ${PROJECT_SOURCE_DIR}/s2client-api/include
+    ${PROJECT_SOURCE_DIR}/s2client-api/contrib/protobuf/src
+    ${PROJECT_BINARY_DIR}/s2client-api/generated
+    ${PROJECT_SOURCE_DIR}/rapidjson
+	if (WIN32) # Curl include for windows
+		include_directories(SYSTEM
+				${PROJECT_SOURCE_DIR}/curl/include/curl
+		)
+	endif ()
+)
+
+# Link directories
+link_directories(${PROJECT_BINARY_DIR}/s2client-api/bin)
+
+if (WIN32) # Curl library for windows
+	link_directories(${PROJECT_SOURCE_DIR}/curl/lib)
+endif ()
+
+# Create the executable.
+add_executable(Sc2LadderServer ${SOURCES_SC2LADDERSERVER})
+target_link_libraries(Sc2LadderServer
+    sc2api sc2lib sc2utils sc2protocol civetweb libprotobuf
+)
+if (WIN32) # Platform specific curl library
+	target_link_libraries(Sc2LadderServer libcurl.lib)
+elseif (UNIX)
+	target_link_libraries(Sc2LadderServer curl)
```

Comment:
Well, if you enforce link with curl you should add the 'find curl' routine/command too.

Reply:
Sorry @alkurbatov just noticed I hadn't replied to this.
I only have a rough idea of what FindCURL does exactly and what it means for the build.
Is it something you think could be addressed after the merge in cleanup, or should be addressed now?
I'm also cognizant of the possibility of moving from CURL to something else.